src/nodeTypes.ml

```diff
@@ -0,0 +1,28 @@
+module ArrayBuffer = struct
+  type t
+end
```

Comment:
This will cause cyclic dependency issues again if the modules need to refer to each others types. But it's a pretty simple refactor to resolve at least.